Fix warnings during build

> g++ -c -Wall main.cpp -o main.o
In file included from Approx.h:18,
g++ -c -Wall Approx.cpp -o Approx.o
                 from main.cpp:7:
Engine.h: In constructor ‘CUnSize::CUnSize(double, double, double)’:
Engine.h:172:29: warning: ‘CUnSize::a2’ will be initialized after [-Wreorder]
  172 |  double a,m,tau,alpha,beta, a2,tau2;
      |                             ^~
Engine.h:172:11: warning:   ‘double CUnSize::m’ [-Wreorder]
  172 |  double a,m,tau,alpha,beta, a2,tau2;
      |           ^
Engine.h:147:2: warning:   when initialized here [-Wreorder]
  147 |  CUnSize(double a=1, double m=1, double tau=1): a(a),a2(a*a),m(m),tau(tau),tau2(tau*tau),alpha(m*pow(a/tau,2)),
      |  ^~~~~~~
Engine.h:172:32: warning: ‘CUnSize::tau2’ will be initialized after [-Wreorder]
  172 |  double a,m,tau,alpha,beta, a2,tau2;
      |                                ^~~~
Engine.h:172:17: warning:   ‘double CUnSize::alpha’ [-Wreorder]
  172 |  double a,m,tau,alpha,beta, a2,tau2;
      |                 ^~~~~
Engine.h:147:2: warning:   when initialized here [-Wreorder]
  147 |  CUnSize(double a=1, double m=1, double tau=1): a(a),a2(a*a),m(m),tau(tau),tau2(tau*tau),alpha(m*pow(a/tau,2)),
      |  ^~~~~~~
In file included from Approx.h:18,
                 from main.cpp:7:
Engine.h: In constructor ‘CLattice::CLattice(double, double (*)(double), double (*)(double))’:
Engine.h:198:30: warning: ‘CLattice::dV_dr’ will be initialized after [-Wreorder]
  198 |  double L,L_2,(*V)(double),(*dV_dr)(double);
      |                              ^~~~~
Engine.h:197:6: warning:   ‘int CLattice::N’ [-Wreorder]
  197 |  int N;
      |      ^
In file included from Approx.h:18,
                 from main.cpp:7:
Engine.h:178:2: warning:   when initialized here [-Wreorder]
  178 |  CLattice(double L, double (*V)(double), double (*dV_dr)(double)): L(L),L_2(L/2.),V(V),dV_dr(dV_dr),N(0) {};
      |  ^~~~~~~~
In file included from Approx.h:18,
g++ -c -Wall Engine.cpp -o Engine.o
                 from Approx.cpp:6:
Engine.h: In constructor ‘CUnSize::CUnSize(double, double, double)’:
Engine.h:172:29: warning: ‘CUnSize::a2’ will be initialized after [-Wreorder]
  172 |  double a,m,tau,alpha,beta, a2,tau2;
      |                             ^~
Engine.h:172:11: warning:   ‘double CUnSize::m’ [-Wreorder]
g++ -c -Wall Equation.cpp -o Equation.o
  172 |  double a,m,tau,alpha,beta, a2,tau2;
Engine.h:197:6: warning:   ‘int CLattice::N’ [-Wreorder]
  197 |  int N;
      |      ^
In file included from Lattice.cpp:5:
Engine.h:178:2: warning:   when initialized here [-Wreorder]
  178 |  CLattice(double L, double (*V)(double), double (*dV_dr)(double)): L(L),L_2(L/2.),V(V),dV_dr(dV_dr),N(0) {};
      |  ^~~~~~~~
g++ -c -Wall Polynomial.cpp -o Polynomial.o
Polynomial.cpp: In function ‘double polChebishev_1(int, double)’:
Polynomial.cpp:45:19: warning: unused variable ‘N’ [-Wunused-variable]
   45 |   default: double N = (double)(--n);
      |                   ^
Polynomial.cpp: In function ‘double polChebishev_2(int, double)’:
Polynomial.cpp:57:19: warning: unused variable ‘N’ [-Wunused-variable]
   57 |   default: double N = (double)(--n);
      |                   ^
g++ -c -Wall Random.cpp -o Random.o
g++ -c -Wall TabFunc.cpp -o TabFunc.o
In file included from TabFunc.h:16,
                 from TabFunc.cpp:5:
Engine.h: In constructor ‘CUnSize::CUnSize(double, double, double)’:
Engine.h:172:29: warning: ‘CUnSize::a2’ will be initialized after [-Wreorder]
  172 |  double a,m,tau,alpha,beta, a2,tau2;
      |                             ^~
Engine.h:172:11: warning:   ‘double CUnSize::m’ [-Wreorder]
  172 |  double a,m,tau,alpha,beta, a2,tau2;
      |           ^
Engine.h:147:2: warning:   when initialized here [-Wreorder]
  147 |  CUnSize(double a=1, double m=1, double tau=1): a(a),a2(a*a),m(m),tau(tau),tau2(tau*tau),alpha(m*pow(a/tau,2)),
      |  ^~~~~~~
Engine.h:172:32: warning: ‘CUnSize::tau2’ will be initialized after [-Wreorder]
  172 |  double a,m,tau,alpha,beta, a2,tau2;
      |                                ^~~~
Engine.h:172:17: warning:   ‘double CUnSize::alpha’ [-Wreorder]
  172 |  double a,m,tau,alpha,beta, a2,tau2;
      |                 ^~~~~
Engine.h:147:2: warning:   when initialized here [-Wreorder]
  147 |  CUnSize(double a=1, double m=1, double tau=1): a(a),a2(a*a),m(m),tau(tau),tau2(tau*tau),alpha(m*pow(a/tau,2)),
      |  ^~~~~~~
In file included from TabFunc.h:16,
                 from TabFunc.cpp:5:
Engine.h: In constructor ‘CLattice::CLattice(double, double (*)(double), double (*)(double))’:
Engine.h:198:30: warning: ‘CLattice::dV_dr’ will be initialized after [-Wreorder]
  198 |  double L,L_2,(*V)(double),(*dV_dr)(double);
      |                              ^~~~~
Engine.h:197:6: warning:   ‘int CLattice::N’ [-Wreorder]
  197 |  int N;
      |      ^
In file included from TabFunc.h:16,
                 from TabFunc.cpp:5:
Engine.h:178:2: warning:   when initialized here [-Wreorder]
  178 |  CLattice(double L, double (*V)(double), double (*dV_dr)(double)): L(L),L_2(L/2.),V(V),dV_dr(dV_dr),N(0) {};
      |  ^~~~~~~~
g++  main.o Approx.o Engine.o Equation.o Interval.o Lattice.o Polynomial.o Random.o TabFunc.o -o app

--- a/Engine.h
+++ b/Engine.h
@@ -169,7 +169,7 @@ public:
 	double P_u_to_s(double d=1) { return m/(a*tau2)*d; };
 	
 private:
-	double a,m,tau,alpha,beta, a2,tau2;
+	double a,a2,m,tau,tau2,alpha,beta;
 };
 
 class CLattice
@@ -194,8 +194,8 @@ public:
 	
 	double getRecommendedTau(double gamma=0.6);
 
-	int N;
 	double L,L_2,(*V)(double),(*dV_dr)(double);
+	int N;
 	CMatrix XYZ,VXYZ;
 
 private:

--- a/Polynomial.cpp
+++ b/Polynomial.cpp
@@ -42,8 +42,8 @@ double polChebishev_1(int n, double x)
 		case 0: return 1.0; break;
 		case 1: return x; break;
 
-		default: double N = (double)(--n);
-			return 2.0*x*polChebishev_1(n,x) - polChebishev_1(n-1,x);
+		default: 
+			return 2.0*x*polChebishev_1(n-1,x) - polChebishev_1(n-2,x);
 	}
 }
 
@@ -54,7 +54,7 @@ double polChebishev_2(int n, double x)
 		case 0: return 1.0; break;
 		case 1: return 2.0*x; break;
 
-		default: double N = (double)(--n);
-			return 2.0*x*polChebishev_2(n,x) - polChebishev_2(n-1,x);
+		default: 
+			return 2.0*x*polChebishev_2(n-1,x) - polChebishev_2(n-2,x);
 	}
 }

